ivadomed/inference.py

```diff
@@ -345,6 +345,10 @@ def segment_volume(folder_model: str, fname_images: list, gpu_id: int = 0, optio
 
     kernel_3D = bool('Modified3DUNet' in context and context['Modified3DUNet']['applied']) or \
                 not context['default_model']['is_2d']
+    length_2D = context["default_model"]["length_2D"] if "length_2D" in context["default_model"] else []
+    stride_2D = context["default_model"]["stride_2D"] if "stride_2D" in context["default_model"] else []
```

Comment:
Would it work if the selected model is not the "default_model"? Or are we enforcing that these params are only defined in the `default_model`?

Reply:
We had this discussion previously [here](https://github.com/ivadomed/ivadomed/pull/763#discussion_r616367317), and decided to go with `length_2D` and `stride_2D` in the `default_model`.
From the [architecture settings](https://ivadomed.org/en/latest/configuration_file.html#architecture) in the doc, there are 4 architectures: default, Filmed, Hemis and 3D. Hemis and 3D do not apply in this case. And if I understand correctly, the FilmedUnet settings are applied "on top" of the default model?
Maybe I'm wrong, I feel like I'm missing something here. Can you give me an example of model where we would want to use the patch that is not the default_model? Thanks!

Reply:
>

eheh you are very right :-)

> And if I understand correctly, the FilmedUnet settings are applied "on top" of the default model?

Yes absolutely, it is done [here](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/main.py#L156).

All good then!!